lib/http/request.ts

```diff
@@ -113,7 +128,8 @@ export function httpRequest(sdk: OktaAuthHttpInterface, options: RequestOptions)
       withCredentials = options.withCredentials === true, // default value is false
       storageUtil = sdk.options.storageUtil,
       storage = storageUtil!.storage,
-      httpCache = sdk.storageManager.getHttpCache(sdk.options.cookies);
+      httpCache = sdk.storageManager.getHttpCache(sdk.options.cookies),
+      canRetry = options.canRetry;
```

Comment:
Can you add an additional option here to gate this logic? Something like `pollingIntent: true/false`. Then change L163 to

```js
if (pollingIntent && isIOS() { ... }
```

Reply:
Both the poll function of `authn` and the SIW polling methods would need to pass this flag

Reply:
Done